File: pattern.cpp
// 1) for the outer loops, count the number of lines
// 2) for the inner loops, focus on the columns and connect them to the rows
// 3) print inside the inner loop
// 4) observer symmetry  [optional]

#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
using namespace std;

void pattern1(int n)
{
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
            cout << "*";
        }
        cout << endl;
    }
}

int main()
{
    int n;
    cin >> n;
    pattern1(n);
    return 0;
}

File: pattern.js
const n = 5; // Number of rows

for (let i = 1; i <= n; i++) {
  // Outer loop for rows
  let row = "";
  for (let j = 1; j <= n - i + 1; j++) {
    // Inner loop for columns (stars in each row)
    row += j + " "; // Append the current number to the row string
  }
  console.log(row);
}
